Analysis/include/Luau/Error.h

```diff
@@ -526,6 +526,36 @@ struct GenericBoundsMismatch
     bool operator==(const GenericBoundsMismatch& rhs) const;
 };
 
+// Used `f<<T>>` where f is not a function
+struct ExplicitlySpecifiedGenericsOnNonFunction
+{
+    enum class InterestingEdgeCase
+    {
+        None,
+        MetatableCall,
+        Intersection,
+    };
+
+    InterestingEdgeCase interestingEdgeCase;
```

Comment:
Can we also put the type that we attempted to instantiate generics on here?

Reply:
As in, keep the rest of the code not using it, but include it as a field on the struct?

Reply:
Yeah, that would be fine. I think it probably belongs in the error message too, but we'll probably want to play with it more to see if that's actually true. Either way, will make it easier to test and debug anything interacting with intersection.